logging/api/export.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+from gcloud import logging
+from oauth2client.client import GoogleCredentials
+
+SINK_NAME = 'mysink'
+
+
+def create_sink_if_not_exists(client, project_id, destination_bucket,
+                              sink_name):
+    """Creates a logging sink"""
+    # [START create]
+    sink = client.sink(sink_name,
+                       'logName="projects/{}/logs/syslog" '
+                       'AND severity>=ERROR'
+                       .format(project_id),
+                       'storage.googleapis.com/{}'.format(destination_bucket))
+    if not sink.exists():
+        sink.create()
+        print("Created sink {}".format(sink.name))
+    # [END create]
+    return sink
+
+
+def list_sinks(client):
+    """Lists all sinks for a project"""
+    print("Listing sinks available")
+
+    # [START list]
+    sinks = []
+    while True:
+        new_sinks, token = client.list_sinks()
+        sinks += new_sinks
+        if token is None:
+            break
+
+    for sink in sinks:
+        print('{}: {}'.format(sink.name, sink.destination))
+    # [END list]
+
+    return sinks
+
+
+def update_sink(sink):
+    """Changes the filter of a sink"""
```

Comment:
Sink filter? (seriously I have no idea what this means)

![](http://g03.a.alicdn.com/kf/HTB1OaQsKpXXXXabaXXXq6xXFXXXd/Brief-Mesh-Stainless-Steel-Kitchen-font-b-Bathroom-b-font-font-b-Sink-b-font-Strainer.jpg)